mlrun/projects/project.py

```diff
@@ -35,50 +32,117 @@
 from ..utils.clones import clone_git, clone_tgz, clone_zip, get_repo_url
+from .pipelines import (
+    FunctionsDict,
+    WorkflowSpec,
+    _PipelineRunStatus,
+    get_db_function,
+    get_workflow_engine,
+)
 
 
 class ProjectError(Exception):
     pass
 
 
-def new_project(name, context=None, init_git=False, user_project=False):
-    """Create a new MLRun project
+def init_repo(context, url, init_git):
+    repo = None
+    if not path.isdir(context):
+        raise ValueError(f"context {context} is not an existing dir path")
+    try:
+        repo = Repo(context)
+        url = get_repo_url(repo)
+    except Exception:
+        if init_git:
+            repo = Repo.init(context)
+    return repo, url
+
+
+def new_project(
+    name,
+    context=None,
+    init_git=False,
+    user_project=False,
+    remote=None,
+    from_template=None,
+    secrets=None,
+    description=None,
+) -> "MlrunProject":
+    """Create a new MLRun project, optionally load it from a yaml/zip/git template
+
+    example::
+
+        # create a project with local and marketplace functions, a workflow, and an artifact
+        project = mlrun.new_project("myproj", "./", init_git=True, description="my new project")
+        project.set_function('prep_data.py', 'prep-data', image='mlrun/mlrun', handler='prep_data')
+        project.set_function('hub://sklearn_classifier', 'train')
+        project.set_artifact('data', Artifact(target_path=data_url))
+        project.set_workflow('main', "./myflow.py")
+        project.save()
+
+        # run the "main" workflow (watch=True to wait for run completion)
+        project.run("main", watch=True)
+
+    example (load from template)::
+
+        # create a new project from a zip template (can also use yaml/git templates)
+        # initialize a local git, and register the git remote path
+        project = mlrun.new_project("myproj", "./", init_git=True,
+                                    remote="git://github.com/mlrun/project-demo.git",
+                                    from_template="http://mysite/proj.zip")
+        project.run("main", watch=True)
+
 
     :param name:         project name
     :param context:      project local directory path
     :param init_git:     if True, will git init the context dir
     :param user_project: add the current user name to the provided project name (making it unique per user)
+    :param remote:       remote Git url
+    :param from_template:     path to project YAML/zip file that will be used as from_template
+    :param secrets:      key:secret dict or SecretsStore used to download sources
+    :param description:  text describing the project
 
     :returns: project object
     """
-    if user_project:
-        user = environ.get("V3IO_USERNAME") or getpass.getuser()
-        name = f"{name}-{user}"
-
-    project = MlrunProject(name=name)
+    context = context or "./"
+    name = _user_project_name(name, user_project)
```

Comment:
```suggestion
    name = _add_username_to_project_name_if_needed(name, user_project)
```

Reply:
the fact we concat is a current implementation, we may want to change later, better to use a name representing it e.g. `_enrich_project_name_with_user_if_needed`

Reply:
done